rct229/rulesets/ashrae9012019/section18/section18rule2.py

```diff
@@ -0,0 +1,290 @@
+            id="18-2",
+            description="Does the modeled system serve the appropriate zones (one system per zone for system types 1, 2, 3, 4, 9, 10, 11, 12, and 13 and "
+            "one system per floor for system types 5, 6, 7, and 8, with the exception of system types 5 or 7 serving laboratory spaces - these systems should serve ALL laboratory zones in the buidling).",
+            ruleset_section_title="HVAC",
+            standard_section="Section 18 HVAC_SystemZoneAssignment",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    def create_data(self, context, data):
+        rmd_b = context.baseline
+        rmd_p = context.proposed
+        climate_zone_b = data["climate_zone"]
+
+        baseline_system_types_dict_b = get_baseline_system_types(rmd_b)
+        applicable_hvac_sys_ids_b = [
+            hvac_id
+            for sys_type in baseline_system_types_dict_b
+            for target_sys_type in APPLICABLE_SYS_TYPES
+            if baseline_system_type_compare(sys_type, target_sys_type, False)
+            for hvac_id in baseline_system_types_dict_b[sys_type]
+        ]
+
+        zones_and_terminal_unit_list_dict_b = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmd_b)
+        )
+        lab_zone_hvac_systems = get_lab_zone_hvac_systems(
+            rmd_b, rmd_p, climate_zone_b, False
+        )  # TODO the last arg is leap_year
+        lab_zone_exhaust_b = get_building_total_lab_exhaust_from_zone_exhaust_fans(
+            rmd_b
+        )
+
+        # create a hvac data dict
+        hvac_data_b = {}
+        for hvac_id_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            rmd_b,
+        ):
+            if hvac_id_b in applicable_hvac_sys_ids_b:
+                hvac_data_b[hvac_id_b] = {
+                    "is_zone_single_zone_sys_b": any(
+                        [
+                            sys_type in APPLICABLE_SYS_TYPES
+                            for sys_type in APPLICABLE_SYS_TYPES
+                        ]
+                    ),
+                    "does_sys_only_serve_lab_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_zones_only"]
+                        and lab_zone_exhaust_b > AIRFLOW_15000_CFM
+                        and len(lab_zone_hvac_systems["lab_zones_only"]) == 1
+                    ),
+                    "does_sys_serve_lab_b": (
+                        hvac_id_b in lab_zone_hvac_systems["lab_zones_only"]
+                        and lab_zone_exhaust_b <= AIRFLOW_15000_CFM
```

Comment:
I am not sure if this logic needs `lab_zone_exhaust_b <= AIRFLOW_15000_CFM`

Reply:
I believe ```lab_zone_exhaust_b``` is needed because when result's pass/faill is determined there is ```if lab_zone_exhaust  > 15000``` condition. Please let me know if I am misunderstanding something.